File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        map = defaultdict(int)
        for char in s:
            map[char] += 1

        for char in t:
            if char not in map : return False
            if not map[char] : return False
            map[char] -= 1
        if len(s) != len(t) : return False
        
        return True

File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        vector<int> sM(26, 0);
        vector<int> tM(26, 0);
        for(auto &c : s) {
            sM[c - 'a']++;
        }
        for(auto &c : t) {
            tM[c - 'a']++;
        }
        return sM == tM;
    }
};